core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala

```diff
@@ -47,7 +48,24 @@ private[spark] case class ResourceRequest(
     discoveryScript: Option[String],
     vendor: Option[String])
 
-private[spark] case class ResourceRequirement(resourceName: String, amount: Int)
+/**
+ * Case class that represents a user's resource requirement as given by configuration
+ * (e.g spark.task.resource.[resource type].amount = 4)
```

Comment:
nit: do we use `[resource type]` elsewhere? I'd prefer keep the naming similar to other parts of docs/comments, e.g. `${resourceName}`